sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_methods.c

```diff
@@ -0,0 +1,487 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "test_az_iot_hub_client.h"
+#include <az_iot_hub_client.h>
+#include <az_span.h>
+
+#include <az_precondition_internal.h>
+#include <az_precondition.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <cmocka.h>
+#include <az_test_precondition.h>
+
+#define TEST_SPAN_BUFFER_SIZE 128
+
+#define TEST_DEVICE_ID_STR "my_device"
+#define TEST_DEVICE_HOSTNAME_STR "myiothub.azure-devices.net"
+#define TEST_MODULE_ID "my_module_id"
```

Comment:
I don't think this is being used

Reply:
Would some of these api's change if the module id is set? Are tests needed?

Reply:
Indeed itś not being used. Removed.
Well, direct methods do not depend on the device/module identity, so no they would not change unless the hub changes their design.
Adding a test here for that would be P4 I would say, since there is no impact right now. A test here would be just for eventual future changes, so it doesn't make sense to add anything like that now.